lib/relation-definition.js

```diff
@@ -1325,7 +1362,8 @@ BelongsTo.prototype.related = function (refresh, params) {
       if (query.where[pk] === undefined
         || query.where[pk] === null) {
         // Foreign key is undefined
-        return process.nextTick(cb);
+        process.nextTick(cb);
+        return cb.promise;
```

Comment:
I find that rather ugly. Why does `related()` need to know implementation details of `get`? I'd much rather keep this knowledge inside `get()`.

```js
var cb = cb || utils.createPromiseCallback();
this.related(true, cb);
return cb.promise;
```